tests/sql/test_sql.py

```diff
@@ -320,3 +320,23 @@ def test_cast_image():
     df = daft.from_pydict({"img": s})
     actual = daft.sql("select cast(img as image(RGB)) from df", **{"df": df}).collect()
     assert actual.schema()["img"].dtype == DataType.image("RGB")
+
+
+def test_count_pushdown(capsys):
+    data = [
+        {"id_spec": 1, "age": "2020-01-15", "tags": ["a", "b"]},
+        {"id_spec": 2, "age": "2020-01-16", "tags": ["c"]},
+        {"id_spec": 3, "age": None, "tags": None},
+    ]
+
+    df = daft.from_pylist(data)
+
+    result_df = daft.sql("SELECT count(1) as total FROM df", **{"df": df})
+    result = result_df.collect().to_pydict()
+
+    assert result == {"total": [3]}, "count(1) return 3 rows"
+
+    result_df.explain(show_all=True)
+    captured = capsys.readouterr()
+
+    assert "id_spec" in captured.out, "id_spec should be in the pushdown"
```

Comment:
Maybe we should check for the string "count(col(id_spec)" since all the column names are currently in the plan.